components/core/src/clp_s/search/sql/sql.cpp

```diff
@@ -0,0 +1,71 @@
+#include <any>
+#include <iostream>
+#include <string>
+
+#include <antlr4-runtime.h>
+#include <spdlog/spdlog.h>
+
+#include "../EmptyExpr.hpp"
+#include "SqlBaseVisitor.h"
+#include "SqlLexer.h"
+#include "SqlParser.h"
+
+using namespace antlr4;
+using namespace sql;
+
+namespace clp_s::search::sql {
+class ErrorListener : public BaseErrorListener {
+public:
+    void syntaxError(
+            Recognizer* recognizer,
+            Token* offending_symbol,
+            size_t line,
+            size_t char_position_in_line,
+            std::string const& msg,
+            std::exception_ptr e
+    ) override {
+        m_error = true;
+        m_error_message = msg;
+    }
+
+    bool error() const { return m_error; }
+
+    std::string const& message() const { return m_error_message; }
+
+private:
+    bool m_error{false};
+    std::string m_error_message;
+};
+
+class ParseTreeVisitor : public SqlBaseVisitor {
+public:
+    std::any visitStart(SqlParser::StartContext* ctx) override { return EmptyExpr::create(); }
+};
```

Comment:
Sorry to miss this in the last round of review. Shall we move these two classes into the anon namespace? Or if there's plan to reuse them in other places, we should probably move them into the dedicated headers.

Reply:
I'll move the ErrorListener into its own header, and put the visitor classes for sql and kql into the anonymous namespace.

I was worried that BaseErrorListener was part of the generated code for each grammar but it's part of the generic antlr runtime so there shouldn't be any issues.